restored role-check before hint to create mupro

wolfgangroese opened the pull request:
not checking the users role caused error "forbidden" when user is admin-role only as musician profile (meService) can only get by performer.

wolfgangroese committed:
restored role-check before hint to create mupro

VILLAN3LL3 commented on src/app/features/dashboard/dashboard.component.ts:
```diff
@@ -115,18 +115,24 @@ export class DashboardComponent implements OnInit, AfterViewInit, OnDestroy {
+        filter((roles: string[]) => roles.includes('Performer'))
+      )
+      .subscribe(() => {
+        this.hasMusicianProfile$ = this.meService.getProfilesMusician().pipe(
```
wenn hasMusicianProfile$ eh ein observable ist, brauchst du hier kein subscribe, sondern kannst es an das observable davor pipen

VILLAN3LL3 commented on src/app/features/dashboard/dashboard.component.ts:
```diff
@@ -115,18 +115,24 @@ export class DashboardComponent implements OnInit, AfterViewInit, OnDestroy {
 
     this.menuItems = this.authService.currentUser.pipe(
       map((token) => token!.roles),
-      map((roles) => roles.map((role) => ({ routerLink: [`/arpa/dashboard/${role}`], label: role.toUpperCase() })))
```
Ich finde es ehrlich gesagt nicht gut dieses Feature in die dashboard component zu packen. Die dashboard component ist als Grundgerüst für alle Roles gedacht, die spezifischen Implementierungen sind separat. Außerdem gilt dieses Feature nur für Performer. Ich würde deshalb ein neues Widget anlegen, dort die Logik reinbauen und das dann dem performer dashboard zuweisen.

wolfgangroese replied:
Absolut dafür, danke für den Hinweis. Traue mir das nur im Moment noch nicht selbst zu da noch andere Probleme auftauchen wie, daß bspw. "ComponentFactoryResolver" als "deprecated" markiert ist und noch so was in der Art weiter unten. Ich würde das lieber zusammen mit @dmpasilva angehen am Wochenende (falls er Zeit hat) und dann erst das ganze in separates Widget packen, wenn das drumrum auch geklärt ist. "ngComponentOutlet" könnte evtl. die Lösung sein, blick aber noch nicht ganz durch. 😬

VILLAN3LL3 replied:
alles klar. wenn ich was übernehmen soll, gib bescheid

wolfgangroese committed:
Refactor dashboard initialization in dashboard.component.ts

The dashboard initialization code has been refactored for better readability and maintainability. The RxJS operator 'switchMap' has been added to handle async code more efficiently within the 'checkIfUserHasMusicianProfile' method. Additionally, the method 'ngOnInit()' has been cleaned up by extracting its logic into separate, newly created methods.

wolfgangroese committed:
Enhance documentation for DashboardComponent & related classes

Significant updates were made to improve the clarity and detail of documentation within the DashboardComponent file. Notably, descriptions were added to CardLayout interface, ArpaWidgetConfigDirective, and WidgetStateService classes. In addition, annotation for the directive and its usage was provided, and a more comprehensive explanation was attached to the DashboardComponent.

wolfgangroese committed:
Merge branch 'develop' into bug/ARPA-955-put-back-role-condition-in-dashboard-mupro-create-hint

wolfgangroese committed:
Merge branch 'develop' into bug/ARPA-955-put-back-role-condition-in-dashboard-mupro-create-hint
